src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToRunAnAotApp.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using System.Runtime.CompilerServices;
+using System.Text;
+using FluentAssertions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.DependencyModel;
+using Microsoft.NET.Build.Tasks;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using Newtonsoft.Json.Linq;
+using Xunit;
+using Xunit.Abstractions;
+using static Microsoft.NET.Publish.Tests.PublishTestUtils;
+
+namespace Microsoft.NET.Publish.Tests
+{
+    public class GivenThatWeWantToRunAnAotApp : SdkTest
+    {
+        public GivenThatWeWantToRunAnAotApp(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        [RequiresMSBuildVersionTheory("17.0.0.32901")]
+        [InlineData(LatestTfm)]        
+        public void NativeAot_only_runs_when_switch_is_enabled(string targetFramework)
```

Comment:
Given that there is a file that tests Aot apps I will include it there instead and verify that the warnings are generated 
https://github.com/dotnet/sdk/blob/main/src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishAnAotApp.cs